devices/Ld2410/Ld2410/Ld2410.cs

```diff
@@ -0,0 +1,379 @@
+using System.IO.Ports;
+using System.Threading;
+
+using Ld2410.Commands;
+using Ld2410.Reporting;
+
+namespace Ld2410
+{
+	/// <summary>
+	/// LD2410 radar module client.
+	/// </summary>
+	public sealed class Radar : IDisposable
+	{
+		private readonly SerialPort serialPort;
+		private readonly AutoResetEvent onAckReceived;
+
+		/// <summary>
+		/// Measurement event handler delegate.
+		/// </summary>
+		/// <param name="sender">The sender object instance.</param>
+		/// <param name="report">The report frame instance.</param>
+		public delegate void MeasurementEventHandler(object sender, ReportFrame report);
+
+		/// <summary>
+		/// Occurs when a new measurement report has been received from the radar module.
+		/// </summary>
+		public event MeasurementEventHandler OnMeasurementReceived;
+
+		/// <summary>
+		/// Gets the current radar configurations.
+		/// </summary>
+		/// <remarks>
+		/// These configurations are retrieved when <see cref="Radar.Connect"/> is executed.
+		/// You must call <see cref="Radar.ReadConfigurations"/> as needed to refresh this property.
+		/// </remarks>
+		public DeviceConfiguration Configuration { get; }
+
+		/// <summary>
+		/// Gets the current firmware version running on the connected LD2410.
+		/// </summary>
+		public string FirmwareVersion { get; private set; }
+
+		/// <summary>
+		/// Gets the amount of time to wait for a response before abandoning the command execution.
+		/// </summary>
+		public TimeSpan CommandTimeout { get; }
+
+		/// <summary>
+		/// Gets a value indicating whether the radar is currently in configuration mode.
+		/// </summary>
+		public bool ConfigurationModeEnabled { get; private set; }
+
+		/// <summary>
+		/// Gets a value indicating wheter the radar is currently in engineering mode.
+		/// </summary>
+		public bool EngineeringModeEnabled { get; private set; }
+
+		/// <summary>
+		/// Initializes a new instance of <see cref="Radar"/>.
+		/// </summary>
+		/// <param name="serialPortName">The serial port name to use.</param>
+		/// <param name="baudRate">The baud-rate to use.</param>
+		/// <param name="commandTimeout">The command timeout. If not specified, the default is 5 seconds.</param>
+		/// <exception cref="ArgumentNullException"><paramref name="serialPortName"/> was null or empty string.</exception>
+		/// <exception cref="ArgumentOutOfRangeException"><paramref name="baudRate"/> was equal or less than 0.</exception>
+		/// <remarks>
+		/// Initializing the <see cref="Radar"/> is not enough to connect to the radar. You must call <see cref="Radar.Connect"/>
+		/// when ready to start reading measurements and controlling the radar.
+		/// </remarks>
+		public Radar(string serialPortName, int baudRate = 256_000, TimeSpan commandTimeout = default)
+		{
+			if (string.IsNullOrEmpty(serialPortName))
+			{
+				throw new ArgumentNullException();
+			}
+
+			if (baudRate <= 0)
+			{
+				throw new ArgumentOutOfRangeException();
+			}
+
+			this.Configuration = new()
```

Comment:
coding wise, you don't need all the this.
We're not using it except if mandatory.

Reply:
Yup, all gone now :)